### ISSUE TITLE
Softmax needing 2 outputs is not a bug

### ISSUE THREAD

**Issue by Mostafa-Samir** [crowd-course/datascience#4]
For classifiers, whether its binary classification or multi-class,  scikit-neuralnetwork uses a one-hot-encoded representation of the labels [1] with cross-entropy-loss [2].  This requires that the output layer (the softmax layer) to be a probability distribution over all labels, hence the number of the units in the softmax layer needs to be the number of labels. In binary classification, we have 2 labels: 0 and 1, so the expected behavior is for the softmax layer to have 2 units.

[1] [sknn.mlp.Classifier Source Code](https://github.com/aigamedev/scikit-neuralnetwork/blob/eb2e6118073a5ff5c3f02c7483870e01797fe426/sknn/mlp.py#L379-L384)

[2] [sknn.mlp.MultiLayerPerceptron Documentation, Parameters, loss-type](https://scikit-neuralnetwork.readthedocs.io/en/latest/module_mlp.html#sknn.mlp.MultiLayerPerceptron)
